cmd/fleet/calendar_cron.go

```diff
@@ -231,13 +241,27 @@ func processFailingHostExistingCalendarEvent(
 	calendarEvent *fleet.CalendarEvent,
 	host fleet.HostPolicyMembershipData,
 ) error {
-	updatedEvent, updated, err := calendar.GetAndUpdateEvent(
-		calendarEvent, func(bool) string {
-			return generateCalendarEventBody(orgName, host.HostDisplayName)
+	updatedEvent := calendarEvent
+	updated := false
+	now := time.Now()
+
+	// Check the user calendar every 30 minutes (and not every time)
+	// to reduce load on both Fleet and the calendar service.
+	if time.Since(calendarEvent.UpdatedAt) > 30*time.Minute {
+		var err error
+		updatedEvent, _, err = calendar.GetAndUpdateEvent(calendarEvent, func(conflict bool) string {
+			return generateCalendarEventBody(orgName, host.HostDisplayName, conflict)
 		})
-	if err != nil {
-		return fmt.Errorf("get event calendar on db: %w", err)
+		if err != nil {
+			return fmt.Errorf("get event calendar on db: %w", err)
+		}
+		// Even if fields haven't changed we want to update the calendar_events.updated_at below.
+		updated = true
```

Comment:
Can we use Redis to track the last update? Is that better than writing to DB? This change can be done after initial release.